tidy3d/plugins/adjoint/components/simulation.py

```diff
@@ -0,0 +1,337 @@
+    "fwidth_adjoint",
+)
+
+# bandwidth of adjoint source in units of freq0 if no sources and no `fwidth_adjoint` specified
+FWIDTH_FACTOR = 1.0 / 10
+
+
+@register_pytree_node_class
+class JaxSimulation(Simulation, JaxObject):
+    """A :class:`.Simulation` registered with jax."""
+
+    input_structures: Tuple[JaxStructure, ...] = pd.Field(
+        (),
+        title="Input Structures",
+        description="Tuple of jax-compatible structures"
+        " that may depend on differentiable parameters.",
+        jax_field=True,
+    )
+
+    output_monitors: Tuple[Union[ModeMonitor, DiffractionMonitor], ...] = pd.Field(
+        (),
+        title="Output Monitors",
+        description="Tuple of monitors whose data the differentiable output depends on.",
+    )
+
+    grad_monitors: Tuple[FieldMonitor, ...] = pd.Field(
+        (),
+        title="Gradient Field Monitors",
+        description="Tuple of monitors used for storing fields, used internally for gradients.",
+    )
+
+    grad_eps_monitors: Tuple[PermittivityMonitor, ...] = pd.Field(
+        (),
+        title="Gradient Permittivity Monitors",
+        description="Tuple of monitors used for storing epsilon, used internally for gradients.",
+    )
+
+    fwidth_adjoint: pd.PositiveFloat = pd.Field(
+        None,
+        title="Adjoint Frequency Width",
+        description="Custom frequency width to use for 'source_time' of adjoint sources. "
+        "If not supplied or 'None', uses the average fwidth of the original simulation's sources.",
+        units=HERTZ,
+    )
+
+    @pd.validator("output_monitors", always=True)
+    def _output_monitors_single_freq(cls, val):
+        """Assert all output monitors have just one frequency."""
+        for mnt in val:
+            if len(mnt.freqs) != 1:
+                raise AdjointError(
+                    "All output monitors must have single frequency for adjoint feature. "
+                    f"Monitor '{mnt.name}' had {len(mnt.freqs)} frequencies."
+                )
+        return val
+
+    @pd.validator("output_monitors", always=True)
+    def _output_monitors_same_freq(cls, val):
+        """Assert all output monitors have the same frequency."""
+        freqs = [mnt.freqs[0] for mnt in val]
+        if len(set(freqs)) > 1:
+            raise AdjointError(
+                "All output monitors must have the same frequency, "
+                f"given frequencies of {[f'{f:.2e}' for f in freqs]} (Hz) "
+                f"for monitors named '{[mnt.name for mnt in val]}', respectively."
+            )
+        return val
+
+    @pd.validator("subpixel", always=True)
+    def _subpixel_is_on(cls, val):
+        """Assert subpixel is on."""
+        if not val:
+            raise AdjointError("'JaxSimulation.subpixel' must be 'True' to use adjoint plugin.")
+        return val
+
+    @pd.validator("input_structures", always=True)
+    def _no_overlap(cls, val):
```

Comment:
What about overlaps with static structures, and input_structures touching simulation domain boundary?

Reply:
`input_structures` always get laid on top of static structures, so any overlap there is not concerning.

we can handle `input_structures` touching simulation domain without needing validation. Internally, we always integrate only over the surface of the structure that is inside of the simulation domain, performing an intersection operation at the beginning of the integration step using the simulation.bounds